internal/webhook/inferenceservice/mutating_test.go

```diff
@@ -0,0 +1,356 @@
+	if !ok {
+		panic("failed to cast InferenceService to unstructured")
+	}
+
+	if annotations != nil {
+		unstructuredISVC.SetAnnotations(annotations)
+	}
+
+	if len(predictorSpec) > 0 {
+		if err := unstructured.SetNestedMap(unstructuredISVC.Object, predictorSpec, "spec", "predictor"); err != nil {
+			panic("failed to set nested map: " + err.Error())
+		}
+	}
+
+	return unstructuredISVC
+}
+
+func runTestCase(t *testing.T, tc TestCase) {
+	t.Helper()
+	g := NewWithT(t)
+	sch, ctx := setupTestEnvironment(t)
+
+	var cli client.Client
+	if tc.secretType != "" {
+		// Extract secret name from annotations, default to testSecret if not specified
+		secretName := testSecret
+		if tc.annotations != nil {
+			if name, exists := tc.annotations[annotations.Connection]; exists {
+				secretName = name
+			}
+		}
+
+		secret := createTestSecret(secretName, tc.secretNamespace, tc.secretType, tc.secretData)
+		cli = fake.NewClientBuilder().WithScheme(sch).WithObjects(secret).Build()
+	} else {
+		cli = fake.NewClientBuilder().WithScheme(sch).Build()
+	}
+
+	webhook := createWebhook(cli, sch)
+
+	isvc := createTestInferenceService(testInferenceService, testNamespace, tc.annotations, tc.predictorSpec)
+	isvcRaw, err := json.Marshal(isvc)
+	if err != nil {
+		t.Fatalf("failed to marshal InferenceService: %v", err)
+	}
+
+	req := admission.Request{
+		AdmissionRequest: admissionv1.AdmissionRequest{
+			Operation: tc.operation,
+			Namespace: testNamespace,
+			Object: runtime.RawExtension{
+				Raw: isvcRaw,
+			},
+			Kind: metav1.GroupVersionKind{
+				Group:   gvk.InferenceServices.Group,
+				Version: gvk.InferenceServices.Version,
+				Kind:    gvk.InferenceServices.Kind,
+			},
+		},
+	}
+
+	resp := webhook.Handle(ctx, req)
+	g.Expect(resp.Allowed).To(Equal(tc.expectedAllowed))
+
+	if tc.expectedMessage != "" {
+		g.Expect(resp.Result.Message).To(ContainSubstring(tc.expectedMessage))
+	}
+
+	if tc.expectedPatchCheck != nil {
+		g.Expect(tc.expectedPatchCheck(resp.Patches)).To(BeTrue())
+	}
+} // oci-v1 - simple case for new injection without existing secrets
+func hasImagePullSecretsPatch(expectedSecretName string) func([]jsonpatch.JsonPatchOperation) bool {
+	return func(patches []jsonpatch.JsonPatchOperation) bool {
+		for _, patch := range patches {
+			if patch.Path == "/spec/predictor/imagePullSecrets" {
+				// Should be a single secret in the array
+				if secretsList, ok := patch.Value.([]interface{}); ok && len(secretsList) == 1 {
+					if secretMap, ok := secretsList[0].(map[string]interface{}); ok {
+						if name, exists := secretMap["name"]; exists && name == expectedSecretName {
+							return true
+						}
+					}
+				}
+			}
+		}
+		return false
+	}
+} // uri-v1.
+func hasStorageUriPatch(expectedUri ...string) func([]jsonpatch.JsonPatchOperation) bool {
+	return func(patches []jsonpatch.JsonPatchOperation) bool {
+		expectedValue := ""
+		if len(expectedUri) > 0 {
+			expectedValue = expectedUri[0]
```

Comment:
this always takes the first element, maybe the `expectedUri` should just be a string instead if a vararg ?